lib/state/session_provider.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:sun_flutter_capstone/models/user.dart';
+
+final defaultUser = User(id: 0, username: 'User');
+
+final sessionProvider = StateProvider((_) => defaultUser);
+
+void setSessionUsername(WidgetRef ref, String username) {
+  ref.read(sessionProvider.state).update(
+        (state) => User(id: state.id, username: username),
+      );
+}
```

Comment:
Would it be better to have this kind of indentation
```suggestion
void setSessionUsername(WidgetRef ref, String username) {
  ref.read(sessionProvider.state).update(
    (state) => User(id: state.id, username: username),
  );
}
```

Reply:
The spaces are forced by the Dart formatter the best I can do is this. 😢
```suggestion
void setSessionUsername(WidgetRef ref, String username) {
  ref.read(sessionProvider.state).update(
        (state) => User(
          id: state.id,
          username: username,
        ),
      );
}
```